module/surveys/helper.js

```diff
@@ -1413,6 +1415,103 @@ module.exports = class SurveysHelper {
     });
   }
 
+  /**
+   * List of surveys based on UserId.
+   * @method
+   * @name listSurveyByUserId
+   * @param {String} requestUserId - id of the user to fetch docmunet.
+   * @param {String} queryType -     type of collection to query whether its survey or obs
+   * @returns {JSON} List of surveys.
+   */
+
+  static listSurveyByUserId(requestUserId, queryType = "") {
+    return new Promise(async (resolve, reject) => {
+      try {
+        let surveyDataofUser = await this.surveyDocuments(
+          { createdBy: requestUserId },
+          ["_id", "solutionId", "userId", "name", "createdBy"]
+        );
+
+        if (!(surveyDataofUser.length > 0)) {
+          return resolve({
+            status: httpStatusCode.bad_request.status,
+            message: messageConstants.apiResponses.SURVEY_NOT_FOUND,
+          });
+        }
+        let surveyListedBasedOnStatus = [];
+        let surveySolutionIds = [];
+
+        if (queryType != "") {
+          if (surveyDataofUser.length > 0) {
+            surveySolutionIds = surveyDataofUser.map(function (obj) {
+              return obj.solutionId;
+            });
+          }
+          let surveySubmissionFindQuery = {};
+          if (surveySolutionIds.length > 0) {
+            surveySubmissionFindQuery = {
+              solutionId: { $in: surveySolutionIds },
+              createdBy: requestUserId,
+            };
+          }
+          let surveySubmissionData =
+            await surveySubmissionsHelper.surveySubmissionDocuments(
```

Comment:
@praveenKDass can use aggregate function for both query

Reply:
Changed as aggregate Function @ankit1shahu